### PR TITLE
hangman game code

### PR DESCRIPTION
**Pull request by BaekMinHyeon**
I made a hangman in the issue.
I'd appreciate it if you could give me a feed. thanks

If the code is satisfactory to you, I will also create test code and documentation 

ps. If you want an answer to Exercise 4, I'll post an example answer in the comments as well.

### CODE REVIEW

File: freegames/hangman.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    goto(x, y)
+    screen_write(letter)
+
+
+def draw(x, y, angle, length, straight):
+    """draw a hangman game"""
+    goto(x, y)
+    pendown()
+    left(angle)
+    if straight == True:
```
Comment (grantjenks) [inline freegames/hangman.py:28]:
```suggestion
    if straight:
```

File: freegames/hangman.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        forward(length)
+    elif straight == False:
```
Comment (grantjenks) [inline freegames/hangman.py:30]:
```suggestion
    else:
```

File: freegames/hangman.py [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+        circle(length, None, 100)
+    penup()
+
+def drawHangman(num):
```
Comment (grantjenks) [inline freegames/hangman.py:34]: Use snake_case like `draw_hangman`

File: freegames/hangman.py [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    if num == 1: 
+        """draw head"""
```
Comment (grantjenks) [inline freegames/hangman.py:36]: Use comments, not strings like `# Draw head`. Same for other strings in this function.

File: freegames/hangman.py (hunk 3 above)
Comment (grantjenks) [inline freegames/hangman.py:34]: Change `num` to `step` to reflect that each of these is the next step in drawing the overall picture.

File: freegames/hangman.py [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+        draw(-45, 90, 60, 40, True)
+
+hideturtle()
+speed(0)
+
+fruitword = ['apple', 'avocado', 'banana', 'blueberry', 'cherry', 'citron',
+        'coconut', 'durian', 'fig', 'grape', 'grapefruit', 'kiwi', 'lemon',
+        'lime', 'mandarin', 'mango', 'melon', 'orange', 'papaya', 'peach',
+        'pear', 'persimmon', 'pineapple', 'plum', 'raspberry', 'strawberry',
+        'tomato', 'watermelon']
```
Comment (grantjenks) [inline freegames/hangman.py:61]: Start with just 5 or 10 words. Keep it to 1-2 lines.

File: freegames/hangman.py [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+for i in word:
+    screen_write("_ ")
+correct = []
+use = []
+wrong = 0
+while wrong < 6:
+    """Modify the code to solve Exercise 4"""
+    letter = textinput("Hangman","Guess a letter:")
+    while len(letter) != 1:
+        letter = textinput("Hangman","Guess a letter:")
```
Comment (grantjenks) [inline freegames/hangman.py:80]: Rather than using `textinput`, simply respond to key press instead.

File: freegames/hangman.py [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+    goto(-100,-35)
+    if letter not in use:
```
Comment (grantjenks) [inline freegames/hangman.py:82]: Remove this check. Allow someone to guess the same letter twice. It's dumb but it'll just lose a turn.

File: freegames/hangman.py [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+            if i == letter:
+                screen_write(letter.lower() + " ")
+                correct += letter
+            else:
+                screen_write("_ ") 
+        use += letter 
+        if letter not in word:
+            wrong += 1
+            drawHangman(wrong)
+        move_write(-200, -100, "You use alphbet: ")
```
Comment (grantjenks) [inline freegames/hangman.py:93]:
```suggestion
        move_write(-200, -100, "Guesses: ")
```

**Review by grantjenks — CHANGES_REQUESTED** [review]
This game is pretty fun. I played a copy of it and enjoyed myself. I left a number of comments. Overall, try to simplify the game further. Remove features if you have to. But keep it fun.

**Comment by BaekMinHyeon** [comment]
Thanks for your feedback, I will reflect your feedback and post it as soon as possible.

File: freegames/hangman.py [hunk 9]
```diff
@@ -0,0 +1,107 @@
+"""hangman, letter guessing game
+
+Exercises
+1. Change Hangman's picture (position, length, etc)
+2. Make a additional dictionary, not a fruit
+3. Convert lowercase letters to uppercase letters
```
[the remark this answers is not in the document]
Comment (BaekMinHyeon) [reply freegames/hangman.py:6]: This means that the words on the screen are lowercase, but students can try to change uppercase

**Comment by BaekMinHyeon** [comment]
I have corrected all the feedback you gave me
And modified to receive input with key press without using testinput

**Comment by BaekMinHyeon** [comment]
I created a docs about hangman

**Pull request merged**